src/data/set/basic.lean

```diff
@@ -1735,39 +1740,42 @@ ext $ by simp [set.prod]
 @[simp] theorem empty_prod : set.prod ∅ t = (∅ : set (α × β)) :=
 ext $ by simp [set.prod]
 
+theorem prod_inter_prod : set.prod s₁ t₁ ∩ set.prod s₂ t₂ = set.prod (s₁ ∩ s₂) (t₁ ∩ t₂) :=
+subset.antisymm
+  (assume ⟨a, b⟩ ⟨⟨ha₁, hb₁⟩, ⟨ha₂, hb₂⟩⟩, ⟨⟨ha₁, ha₂⟩, ⟨hb₁, hb₂⟩⟩)
+  (subset_inter
+    (prod_mono (inter_subset_left _ _) (inter_subset_left _ _))
+    (prod_mono (inter_subset_right _ _) (inter_subset_right _ _)))
+
 theorem insert_prod {a : α} {s : set α} {t : set β} :
   set.prod (insert a s) t = (prod.mk a '' t) ∪ set.prod s t :=
-ext begin simp [set.prod, image, iff_def, or_imp_distrib] {contextual := tt}; cc end
+begin ext ⟨x, y⟩, simp [set.prod, image, iff_def, or_imp_distrib, imp.swap] {contextual := tt} end
```

Comment:
Do we have `prod_singleton`, `singleton_prod`, `prod_union`, and `union_prod`?

Reply:
I added them